src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -108,46 +108,27 @@ function DiscoverDocumentsComponent({
   const documents$ = stateContainer.dataState.data$.documents$;
   const savedSearch = useSavedSearchInitial();
   const { dataViews, capabilities, uiSettings, uiActions } = services;
-  const [
-    query,
-    sort,
-    rowHeight,
-    headerRowHeight,
-    rowsPerPage,
-    grid,
-    columns,
-    index,
-    sampleSizeState,
-  ] = useAppStateSelector((state) => {
-    return [
-      state.query,
-      state.sort,
-      state.rowHeight,
-      state.headerRowHeight,
-      state.rowsPerPage,
-      state.grid,
-      state.columns,
-      state.index,
-      state.sampleSize,
-    ];
-  });
-  const setExpandedDoc = useCallback(
-    (doc: DataTableRecord | undefined) => {
-      stateContainer.internalState.transitions.setExpandedDoc(doc);
-    },
-    [stateContainer]
-  );
-
+  const [query, sort, rowHeight, headerRowHeight, rowsPerPage, grid, columns, sampleSizeState] =
+    useAppStateSelector((state) => {
+      return [
+        state.query,
+        state.sort,
+        state.rowHeight,
+        state.headerRowHeight,
+        state.rowsPerPage,
+        state.grid,
+        state.columns,
+        state.sampleSize,
+      ];
+    });
```

Comment:
let me share a thought that doesn't need to be applied here. We are subscribing to AppState which is synced to URL, AppState is also applied to the savedSearch state. We are subscribing to too much state of various sources in the UI, so this one could be changed to use `useSavedSearch`, so there are "just" dataState, savedSearchState and internalState being used.